communications.c

```diff
@@ -82,10 +85,20 @@ int
 send_command(int interactive, unsigned char *cmd)
 {
 	struct sockaddr_un sun;
-	char *wcmd;
-	char ret[1024];
+	char *wcmd, *bufstart;
+	char ret[BUFSZ+1];
+	char success = 0;
 	size_t len;
-	int fd;
+	ssize_t count;
+	int fd, firstloop;
+	int flags = 0x0;
+	FILE *outf = NULL;
+
+#ifdef SENDCMD_DEBUG
+	pid_t pid = getpid();
+	char *dpfx = xsprintf("send_command_%d", pid);
```

Comment:
You are leaking `dpfx` here, there is no `free(dpfx)`.

Reply:
good catch, thanks!